pkg/networkservice/common/discoverforwarder/server.go

```diff
@@ -67,6 +67,13 @@ func (d *discoverForwarderServer) Request(ctx context.Context, request *networks
 	var forwarderName = loadForwarderName(ctx)
 	var logger = log.FromContext(ctx).WithField("discoverForwarderServer", "request")
 
+	if forwarderName == "" {
+		segments := request.Connection.GetPath().GetPathSegments()
+		if pathIndex := int(request.Connection.GetPath().Index); len(segments) > pathIndex+1 {
+			forwarderName = segments[pathIndex+1].Name
+		}
+	}
+
```

Comment:
@NikitaSkrynnik Please note: in case of death we can down into infinity loop. I.e.

1. `discoverfowarder` gets refresh
2. the previously selected fowrarder dies

actual: `discoverforwarder` never reselects forwarder.
expected: `discoverforwarder` reselects forwarder if it is not registered.

So in case if `forwarderName` is empty we should try to re-use the name, but if it is not possible we should do reselection.